airflow/providers/google/cloud/transfers/azure_fileshare_to_gcs.py

```diff
@@ -0,0 +1,191 @@
+    )
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        share_name: str,
+        directory_name: Optional[str] = None,
+        prefix: str = '',
+        wasb_conn_id: str = 'wasb_default',
+        gcp_conn_id: str = 'google_cloud_default',
+        dest_gcs: str,
+        delegate_to: Optional[str] = None,
+        replace: bool = False,
+        gzip: bool = False,
+        google_impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.share_name = share_name
+        self.directory_name = directory_name
+        self.prefix = prefix
+        self.wasb_conn_id = wasb_conn_id
+        self.gcp_conn_id = gcp_conn_id
+        self.dest_gcs = dest_gcs
+        self.delegate_to = delegate_to
+        self.replace = replace
+        self.gzip = gzip
+        self.google_impersonation_chain = google_impersonation_chain
+
+        if dest_gcs and not self._gcs_object_is_directory(self.dest_gcs):
+            self.log.info(
+                'Destination Google Cloud Storage path is not a valid '
+                '"directory", define a path that ends with a slash "/" or '
+                'leave it empty for the root of the bucket.'
+            )
+            raise AirflowException(
+                'The destination Google Cloud Storage path must end with a slash "/" or be empty.'
+            )
+
+    def execute(self, context):
+        azure_fileshare_hook = AzureFileShareHook(self.wasb_conn_id)
+        files = azure_fileshare_hook.list_files(
+            share_name=self.share_name, directory_name=self.directory_name
+        )
+
+        gcs_hook = GCSHook(
+            google_cloud_storage_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            impersonation_chain=self.google_impersonation_chain,
+        )
+
+        dest_gcs_bucket, dest_gcs_object_prefix = _parse_gcs_url(self.dest_gcs)
+
+        # pylint: disable=too-many-nested-blocks
+        if not self.replace:
+            # if we are not replacing -> list all files in the GCS bucket
+            # and only keep those files which are present in
+            # S3 and not in Google Cloud Storage
+            existing_files_prefixed = gcs_hook.list(dest_gcs_bucket, prefix=dest_gcs_object_prefix)
+
+            existing_files = []
+
+            if existing_files_prefixed:
+                # Remove the object prefix itself, an empty directory was found
+                if dest_gcs_object_prefix in existing_files_prefixed:
+                    existing_files_prefixed.remove(dest_gcs_object_prefix)
+
+                # Remove the object prefix from all object string paths
+                for file in existing_files_prefixed:
+                    if file.startswith(dest_gcs_object_prefix):
+                        existing_files.append(file[len(dest_gcs_object_prefix) :])
+                    else:
+                        existing_files.append(file)
```

Comment:
```suggestion
            if existing_files_prefixed:
                # Remove the object prefix itself, an empty directory was found
                if dest_gcs_object_prefix in existing_files_prefixed:
                    existing_files_prefixed.remove(dest_gcs_object_prefix)

                # Remove the object prefix from all object string paths
                for file in existing_files_prefixed:
                    if file.startswith(dest_gcs_object_prefix):
                        existing_files.append(file[len(dest_gcs_object_prefix) :])
                    else:
                        existing_files.append(file)
```
I think we can remove `if existing_files_prefixed:` and everything will work in the same way? Then number of blosks should also be ok for pylint

Reply:
yes, you're right, fixd